### ISSUE TITLE
BART: how are the spans computed during training

### ISSUE THREAD

**Issue by Avmb** [pytorch/fairseq#1762]
#### What is your question?

I can't find any code for the unsupervised pretraining of BART or mBART, so I'll ask here:

the BART paper says:
"Text Infilling A number of text spans are sampled,with  span  lengths  drawn  from  a  Poisson  distribution(λ= 3).  Each span is replaced with a single [MASK] token.   0-length  spans  correspond  to  the  insertion  of [MASK] tokens. "

How many spans are sampled? Is it just one per sentence or more than one. What is the scale of the Poisson distribution, is it number of tokens or number of tokens / sentence length?

The mBART paper says:
"We mask 35% of the words in each instance by random sampling a span length according to a Poisson distribution (λ= 3.5)"

Does this mean that there is only one span per sentence?

**Comment by yinhanliu** [comment]
@Avmb Thanks for the interest.  I will add a README for both of them. To your question - there are multiple spans (average length is 3) were added as noise, which resulted a 35% input tokens are replaced as [Mask]. Poisson distribution sampled # tokens in a span. It doesn't depend sentence length.
